### PR TITLE
Adds 'Add Milestone' feature

### PR DESCRIPTION
**Pull request by ria-arora**
- Adds add_milestone.gs

### CODE REVIEW

File: project_tracker_template/add_milestone.gs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                      newMilestoneRow);
+  // The first milestone row and a milestone row without tasks above it don't
+  // require ungrouping
+  if (currentMilestoneNumber > 1 && prevTaskCount >= 1) {
+    ungroupMilestone(taskSheet, newMilestoneRow);
+  }
+  // Update spreadsheet values
+  updateSpreadsheet();
+  return true;
+}
```
Comment (paneendra-google) [inline project_tracker_template/add_milestone.gs]: newline at EOF

File: project_tracker_template/add_milestone.gs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  var lastMilestoneColumn = currentMilestoneNumber + 1;
+  // Milestone data starts from Column B onwards in the Summary Sheet
+  var newMilestoneColumn = currentMilestoneNumber + 1;
+  summarySheet.insertColumnAfter(lastMilestoneColumn);
+  // Milestone title
+  var titleCell = summarySheet.getRange(5, newMilestoneColumn);
+  var title = 'Milestone ' + currentMilestoneNumber;
+  titleCell.setValue(title);
+  // Desired Launch Date
+  var desiredLaunchDateCell = summarySheet.getRange(6, newMilestoneColumn);
```
Comment (paneendra-google) [inline project_tracker_template/add_milestone.gs]: use constants for row numbers with meaningful names

**Review by paneendra-google — CHANGES_REQUESTED** [review]

File: project_tracker_template/add_milestone.gs [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+ */
+function addMilestoneSummary(desiredLaunchDate, currentMilestoneNumber,
+                             newMilestoneRow) {
+  var summarySheet =
+      SpreadsheetApp.getActiveSpreadsheet().getSheetByName("Summary");
+  var lastMilestoneColumn = currentMilestoneNumber + 1;
+  // Milestone data starts from Column B onwards in the Summary Sheet
+  var newMilestoneColumn = currentMilestoneNumber + 1;
+  summarySheet.insertColumnAfter(lastMilestoneColumn);
+  const titleRow = 5;
```
Comment (paneendra-google) [inline project_tracker_template/add_milestone.gs]: move const to global scope and name as TITLE_ROW

**Review by paneendra-google — CHANGES_REQUESTED** [review]

**Review by paneendra-google — APPROVED** [review]
Just ensure that const works when you turn off v8 (as far as I remember it doesn't), you might have to change const to var in that case

**Comment by ria-arora** [comment]
I just checked, it's not working.
I should put the variables back in the functions as before and just change their type from 'const' to 'var' then I believe.

**Comment by ria-arora** [comment]
Updated type of variables from const to var

**Review by paneendra-google — APPROVED** [review]

**Pull request merged**